python/cudf/cudf/tests/test_rank.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+import numpy as np
+import pandas as pd
+import pytest
+
+from cudf.core import DataFrame
+from cudf.tests.utils import assert_eq
+
+
+class TestRank:
+    col1 = np.array([5, 4, 3, 5, 8, 5, 2, 1, 6, 6])
+    col2 = np.array([5, 4, np.nan, 5, 8, 5, np.inf, np.nan, 6, -np.inf])
```

Comment:
Can you add a non-standard index as well just to test that we maintain it properly?